persistence/src/commonMain/db_user/com/wire/kalium/persistence/MessageAttachments.sq

```diff
@@ -0,0 +1,72 @@
+import com.wire.kalium.persistence.dao.QualifiedIDEntity;
+import kotlin.Boolean;
+import kotlin.Float;
+import kotlin.Int;
+import kotlin.String;
+import kotlin.collections.List;
+import kotlinx.datetime.Instant;
+
+CREATE TABLE MessageAttachments (
+      asset_id TEXT NOT NULL,
+      asset_version_id TEXT NOT NULL,
+      message_id TEXT NOT NULL,
+      conversation_id TEXT AS QualifiedIDEntity NOT NULL,
+      cell_asset INTEGER AS Boolean NOT NULL,
+      local_path TEXT,
+      content_url TEXT,
+      preview_url TEXT,
+      asset_mime_type TEXT NOT NULL,
+      asset_size INTEGER,
+      asset_path TEXT,
+
+      content_hash TEXT,
+
+      -- metadata fields
+      asset_width INTEGER AS Int,
+      asset_height INTEGER AS Int,
+      asset_duration_ms INTEGER,
+
+      asset_transfer_status TEXT NOT NULL DEFAULT 'NOT_DOWNLOADED',
+
+      FOREIGN KEY (message_id, conversation_id) REFERENCES Message(id, conversation_id) ON DELETE CASCADE ON UPDATE CASCADE,
+      PRIMARY KEY (asset_id, message_id, conversation_id)
```

Comment:
I wonder about performance here.

Maybe it is necessary to add an index with `conversation_id` and `message_id`, _without_ `asset_id`, because of the JOIN in `MessageDetailsView`.

The DataBase needs to be able to quickly find `MessageAttachments` without caring for `asset_id`. _i.e._ "give me all message attachments for this message in this conversation"

`asset_id` is part of the primary key, so _I  guess_ it will be in the way. Or at least it should be the last column specified in the `PRIMARY KEY` statement.

Reply:
this can be solved via ordering the pk

      PRIMARY KEY (conversation_id, message_id, asset_id,)